src/hooks/useCalendar.ts

```diff
@@ -0,0 +1,77 @@
+import {
+  addMonths,
+  eachDayOfInterval,
+  getMonth,
+  getYear,
+  startOfWeek,
+  subMonths,
+  subDays,
+  addDays,
+  lastDayOfWeek,
+} from 'date-fns';
+import { useState } from 'react';
+
+const useCalendar = (): CalendarProps => {
+  const [currentDate, setCurrentDate] = useState(new Date());
+
+  const firstDayOfMonth = () => {
+    return new Date(getYear(currentDate), getMonth(currentDate), 1);
+  };
+
+  const lastDayOfMonth = () => {
+    return new Date(getYear(currentDate), getMonth(currentDate) + 1, 0);
+  };
+
+  const prevMonthDates = () => {
+    return eachDayOfInterval({
+      start: startOfWeek(firstDayOfMonth()),
+      end: subDays(firstDayOfMonth(), 1),
+    });
+  };
+
+  const currMonthDates = () => {
+    return eachDayOfInterval({
+      start: firstDayOfMonth(),
+      end: lastDayOfMonth(),
+    });
+  };
+
+  const nextMonthDates = () => {
+    return eachDayOfInterval({
+      start: addDays(lastDayOfMonth(), 1),
+      end: lastDayOfWeek(lastDayOfMonth()),
+    });
+  };
+
+  const prevMonth = () => {
+    setCurrentDate(subMonths(new Date(getYear(currentDate), getMonth(currentDate), 1), 1));
+  };

```

Comment:
매우 마이너하지만 요렇게 코드 단순화해볼 수 있지 않을까요?

```suggestion
  const prevMonth = () => {
    setCurrentDate(subMonths(new Date(currentDate), 1));
  };
```

Reply:
당시에는 시간 관련하여 00:00:00로 고정시켜 부가 로직을 작성해야겠다 했는데
지금보니 `useCalendar`의 `currMonthDates`를 활용하면 되네요.

감사드립니다, 바로 반영하겠습니다!